### PR TITLE
Fix options list input unfocus

### PR DESCRIPTION
**Pull request by mollfpr**
### Details

The issue is on mWeb when we scroll the view programmatically, the keyboard is dismissed where the focus should stay on the input and the keyboard is stay open as the native app does. Creating a touch listener to define is scroll programmatically or from user drag scroll.

### Fixed Issues

$ https://github.com/Expensify/App/issues/11801   
$ https://github.com/Expensify/App/issues/11801#issuecomment-1278549911

### Tests

1. Login with your account
2. Select Fub menu - Select New Group
3. Type the user in the search field
4. Tap on the user
5. Go back to the search field
6. Search for the second user
7. Verified that the input is still in focus and the keyboard does not dismiss when the scroll is changed.

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Login with your account
2. Select Fub menu - Select New Group
3. Type the user in the search field
4. Tap on the user
5. Go back to the search field
6. Search for the second user
7. Verified that the input is still in focus and the keyboard does not dismiss when the scroll is changed.

- [x] Verify that no errors appear in the JS console

#### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

#### Web

https://user-images.githubusercontent.com/25520267/197224439-b64abffe-8602-4da6-994a-c2d4cc03bd14.mov

#### Mobile Web - Chrome

https://user-images.githubusercontent.com/25520267/197224454-9ad4f50a-d761-406f-ae5d-3f541ae0c491.mov

#### Mobile Web - Safari

https://user-images.githubusercontent.com/25520267/197224463-608acd61-33be-42cf-b7e9-e892c39e030f.mov

#### Desktop

https://user-images.githubusercontent.com/25520267/197224486-660c27ee-9a54-4eff-bf5b-365d8461e0cf.mov

#### iOS

https://user-images.githubusercontent.com/25520267/197224706-96bad8af-b919-4783-a0fc-c91a75e7c727.mov

#### Android

https://user-images.githubusercontent.com/25520267/197224650-c8ca6f91-3b7e-4f47-8839-be0d0c68776e.mov

### CODE REVIEW

**Comment by mollfpr** [comment]
Bump @eVoloshchak 🚀

**Comment by eVoloshchak** [comment]
https://user-images.githubusercontent.com/9059945/197559821-8728ea9b-a823-47e1-8e47-062be699e959.mp4

The keyboard isn't hiding on scroll

Not sure what's the issue, the only change I did is to merge your changes into the latest changes from `main`. Could you please merge `main` and retest?

**Comment by mollfpr** [comment]
@eVoloshchak Yup I'll try it and merge the latest `main`.

**Comment by mollfpr** [comment]
@eVoloshchak I just pulled the latest `main` and retest it again on an Android device and the Android simulator.

**Android device**

https://user-images.githubusercontent.com/25520267/197570153-fd5a4406-3162-4852-974f-ade724c2aae9.mp4

**Android simulator**

https://user-images.githubusercontent.com/25520267/197570612-95a06fe4-5b3e-4b4e-85d7-7a43fd13efdb.mov

**Comment by eVoloshchak** [comment]
> I just pulled the latest `main` and retest it again on an Android device and the Android simulator.

Can confirm it's working for me, reviewing

**Comment by eVoloshchak** [comment]
<details><summary><h4>Screenshots</h4></summary>

**Web**

https://user-images.githubusercontent.com/9059945/197596992-f8373067-17f0-4448-823c-519a4e9c3946.mov

**Mobile Web - Chrome**


https://user-images.githubusercontent.com/9059945/197597199-11f3d0cd-b2cd-4044-83f9-891c833121b8.mp4

**Mobile Web - Safari**

https://user-images.githubusercontent.com/9059945/197597474-6a7ac4ea-e941-49ae-bceb-27d9b3fed302.mov

**Desktop**


https://user-images.githubusercontent.com/9059945/197598064-4e2cfa00-7455-41ec-a4fd-3ea01e9cbea5.mov

iOS


https://user-images.githubusercontent.com/9059945/197598326-e11d9083-207f-4334-ab6f-7e6c2a56fa4c.mov

**Android**


https://user-images.githubusercontent.com/9059945/197600312-29f4065f-f30e-46c6-9015-7c050bd3e56e.mov


</details>

<details><summary><h4>PR Reviewer Checklist</h4></summary>

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

File: src/components/OptionsList/index.js [hunk 1]
```diff
@@ -1,19 +1,68 @@
...
+OptionsList.propTypes = {
+    ...propTypes,
+};
 OptionsList.defaultProps = defaultProps;
-OptionsList.displayName = 'OptionsList';
 
-export default withWindowDimensions(OptionsList);
+export default withWindowDimensions(forwardRef((props, ref) => (
+    // eslint-disable-next-line react/jsx-props-no-spreading
+    <OptionsList innerRef={ref} {...props} />
```
Comment (eVoloshchak) [inline src/components/OptionsList/index.js:67]: let's call it `forwardedRef` so it's consistent with the rest of the app

File: src/components/OptionsList/index.js [hunk 2]
```diff
@@ -1,19 +1,68 @@
...
+
+    userTouchEnd() {
+        this.isUserScreenTouched = false;
+    }
+
+    render() {
+        return (
+            <BaseOptionsList
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {..._.omit(this.props, 'innerRef')}
```
Comment (eVoloshchak) [inline src/components/OptionsList/index.js:47]: do we need to omit the `innerRef`?

**Review by eVoloshchak — CHANGES_REQUESTED** [review]

**Comment by eVoloshchak** [comment]
Also please add some comments in `OptionsList/index.js`'s `componentDidMount` method explaining why this logic with listeners is needed and attach a link to the GitHub issue

**Comment by mollfpr** [comment]
@eVoloshchak Updated PR!

> do we need to omit the innerRef?

I guess we need to omit that `innerRef` or `forwardedRef` right now 'cause we don't need that to pass to the `BaseOptionsList` component but let me know what you're suggesting.

**Comment by mollfpr** [comment]
Bump @eVoloshchak

**Review by eVoloshchak — CHANGES_REQUESTED** [review]

**Comment by eVoloshchak** [comment]
Github's `commit change` feature mushed two comments together, the whole comment should be 

```js
        // According to https://github.com/Expensify/App/issues/11801
        // `keyboardDismissMode` is dismissing the keyboard when the list is scrolled programmatically.
        // In this listener we will set `isUserScreenTouched` when the user is touching the screen
        // and `isUserScreenTouched` will be used to decide whether the list was scrolled by the user
        // or programmatically so that the keyboard would be dismissed only when it was scrolled by the user.
```

**Comment by mollfpr** [comment]
@eVoloshchak Yeah you're right, I fix the comment thanks!

**Comment by eVoloshchak** [comment]
retesting

**Comment by eVoloshchak** [comment]
LGTM
cc: @marcochavezf

**Review by eVoloshchak — DISMISSED** [review]

File: src/components/OptionsList/index.js [hunk 3]
```diff
@@ -1,19 +1,74 @@
...
+    componentDidMount() {
+        if (!canUseTouchscreen()) {
+            return;
+        }
+
+        // According to https://github.com/Expensify/App/issues/11801
+        // `keyboardDismissMode` is dismissing the keyboard when the list is scrolled programmatically.
+        // In this listener we will set `isUserScreenTouched` when the user is touching the screen
+        // and `isUserScreenTouched` will be used to decide whether the list was scrolled by the user
+        // or programmatically so that the keyboard would be dismissed only when it was scrolled by the user.
```
Comment (marcochavezf) [inline src/components/OptionsList/index.js:26]:
```suggestion
        // We're setting `isScreenTouched` in this listener only for web platforms with touchscreen (mWeb) where
        // we want to dismiss the keyboard only when the list is scrolled by the user and not when it's scrolled programmatically.
```

File: src/components/OptionsList/index.js [hunk 4]
```diff
@@ -1,19 +1,74 @@
...
-        keyboardDismissMode={props.isSmallScreenWidth ? 'on-drag' : 'none'}
-    />
-));
+class OptionsList extends Component {
+    constructor(props) {
+        super(props);
 
-OptionsList.propTypes = propTypes;
+        this.userTouchStart = this.userTouchStart.bind(this);
+        this.userTouchEnd = this.userTouchEnd.bind(this);
```
Comment (marcochavezf) [inline src/components/OptionsList/index.js:14]:
```suggestion
        this.touchStart = this.touchStart.bind(this);
        this.touchEnd = this.touchEnd.bind(this);
```

File: src/components/OptionsList/index.js [hunk 5]
```diff
@@ -1,19 +1,74 @@
...
+            return;
+        }
+
+        // According to https://github.com/Expensify/App/issues/11801
+        // `keyboardDismissMode` is dismissing the keyboard when the list is scrolled programmatically.
+        // In this listener we will set `isUserScreenTouched` when the user is touching the screen
+        // and `isUserScreenTouched` will be used to decide whether the list was scrolled by the user
+        // or programmatically so that the keyboard would be dismissed only when it was scrolled by the user.
+        document.addEventListener('touchstart', this.userTouchStart);
+        document.addEventListener('touchend', this.userTouchEnd);
```
Comment (marcochavezf) [inline src/components/OptionsList/index.js:28]:
```suggestion
        document.addEventListener('touchstart', this.touchStart);
        document.addEventListener('touchend', this.touchEnd);
```

File: src/components/OptionsList/index.js [hunk 6]
```diff
@@ -1,19 +1,74 @@
...
+    }
+
+    componentWillUnmount() {
+        if (!canUseTouchscreen()) {
+            return;
+        }
+
+        document.removeEventListener('touchstart', this.userTouchStart);
+        document.removeEventListener('touchend', this.userTouchEnd);
```
Comment (marcochavezf) [inline src/components/OptionsList/index.js:37]:
```suggestion
        document.removeEventListener('touchstart', this.touchStart);
        document.removeEventListener('touchend', this.touchEnd);
```

File: src/components/OptionsList/index.js [hunk 7]
```diff
@@ -1,19 +1,74 @@
...
+    }
+
+    userTouchStart() {
+        this.isUserScreenTouched = true;
```
Comment (marcochavezf) [inline src/components/OptionsList/index.js:41]:
```suggestion
        this.isScreenTouched = true;
```

File: src/components/OptionsList/index.js [hunk 8]
```diff
@@ -1,19 +1,74 @@
...
+    }
+
+    userTouchEnd() {
+        this.isUserScreenTouched = false;
```
Comment (marcochavezf) [inline src/components/OptionsList/index.js:45]:
```suggestion
        this.isScreenTouched = false;
```

File: src/components/OptionsList/index.js [hunk 9]
```diff
@@ -1,19 +1,74 @@
...
+
+    render() {
+        return (
+            <BaseOptionsList
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {..._.omit(this.props, 'forwardedRef')}
+                ref={this.props.forwardedRef}
+                onScroll={() => {
+                    // Only dismiss the keyboard whenever the user scrolls the screen
+                    if (!this.isUserScreenTouched) {
```
Comment (marcochavezf) [inline src/components/OptionsList/index.js:56]:
```suggestion
                    if (!this.isScreenTouched) {
```

**Review by marcochavezf — CHANGES_REQUESTED** [review]
The code looks good! Just rename the functions and the variable to make them a bit clearer

**Comment by mollfpr** [comment]
@marcochavezf Updated!

**Review by marcochavezf — APPROVED** [review]
LGTM, thanks guys!

**Pull request merged**